src/test/regress/expected/window_functions.out

```diff
@@ -0,0 +1,915 @@
+         ->  Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
+               Task Count: 4
+               Tasks Shown: One of 4
+               ->  Task
+                     Node: host=localhost port=57637 dbname=regression
+                     ->  WindowAgg  (cost=2.16..3.40 rows=33 width=48)
+                           ->  GroupAggregate  (cost=2.16..2.82 rows=33 width=16)
+                                 Group Key: users_table.user_id, users_table.value_2
+                                 ->  Sort  (cost=2.16..2.24 rows=33 width=12)
+                                       Sort Key: users_table.user_id, users_table.value_2
+                                       ->  Seq Scan on users_table_1400000 users_table  (cost=0.00..1.33 rows=33 width=12)
+(15 rows)
+
+SELECT
+	user_id,
+	1 + sum(value_1),
+	1 + AVG(value_2) OVER (partition by user_id)
+FROM
+	users_table
+GROUP BY
+	user_id, value_2
+ORDER BY
+	2 DESC, 1
+LIMIT 5;
+ user_id | ?column? |      ?column?      
+---------+----------+--------------------
+       4 |       28 | 3.5000000000000000
+       5 |       24 | 3.5000000000000000
+       2 |       17 | 3.5000000000000000
+       3 |       17 | 4.0000000000000000
+       4 |       17 | 3.5000000000000000
+(5 rows)
+
+-- rank and ordering in the reverse order 
+SELECT
+	user_id,
+	avg(value_1),
+	RANK() OVER (partition by user_id order by value_2)
+FROM
+	users_table
+GROUP BY user_id, value_2
+ORDER BY user_id, value_2 DESC;
+ user_id |          avg           | rank 
+---------+------------------------+------
+       1 |     3.6666666666666667 |    4
+       1 |     2.5000000000000000 |    3
+       1 |     3.0000000000000000 |    2
+       1 |     4.0000000000000000 |    1
+       2 |     1.5000000000000000 |    6
+       2 |     3.2000000000000000 |    5
+       2 |     1.6666666666666667 |    4
+       2 |     3.0000000000000000 |    3
+       2 |     1.3333333333333333 |    2
+       2 |     2.0000000000000000 |    1
+       3 |     2.6666666666666667 |    5
+       3 | 1.00000000000000000000 |    4
+       3 |     3.0000000000000000 |    3
+       3 |     2.4000000000000000 |    2
+       3 | 1.00000000000000000000 |    1
+       4 |     3.5000000000000000 |    6
+       4 |     3.2000000000000000 |    5
+       4 |     3.3333333333333333 |    4
+       4 | 0.00000000000000000000 |    3
+       4 |     3.0000000000000000 |    2
+       4 | 1.00000000000000000000 |    1
+       5 |     3.0000000000000000 |    6
+       5 |     2.3333333333333333 |    5
+       5 |     1.6000000000000000 |    4
+       5 |     2.8750000000000000 |    3
+       5 |     3.2000000000000000 |    2
+       5 |     3.0000000000000000 |    1
+       6 |     1.3333333333333333 |    5
+       6 |     2.0000000000000000 |    4
+       6 |     4.0000000000000000 |    3
+       6 | 1.00000000000000000000 |    2
+       6 |     2.5000000000000000 |    1
+(32 rows)
+
+-- order by in the window function is same as avg(value_1) DESC
+SELECT
```

Comment:
Cool query. But, we should add `LIMIT`  to pushdown `ORDER BY`.  Could you go over the tests to see if we have any other queries like this (e.g., the above is also misses the `LIMIT`)?